Core/Src/main.c

```diff
@@ -87,14 +89,41 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // Initializing Values
+  const uint8_t transmit_val[3] = {0x1, 0x8, 0x0};
+  const uint8_t receive_val[3] = {};
```

Comment:
this value shouldn't be const, because the HAL_SPI_TransmitReceive() function will write the response data from the ADC chip into our MCU